backend/app/repositories/expectation.py

```diff
@@ -0,0 +1,81 @@
+import json
+from typing import Any, Optional
+
+from app.repositories.base import BaseRepository, get_repository
+from app.models.expectation import BaseKwargs, Expectation, type_map
+from app.settings import settings
+
+
+class ExpectationRepository(BaseRepository[Expectation]):
+    model_class = Expectation
+    index = settings.EXPECTATION_INDEX
+
+    def query_by_filter(
+        self,
+        *,
+        datasource_id: Optional[str] = None,
+        dataset_id: Optional[str] = None,
+        suggested: Optional[bool] = None,
+        enabled: Optional[bool] = None,
+        asc: Optional[bool] = False,
+    ) -> list[Exception]:
```

Comment:
Well, it depends 🙃 Starting Python 3.9, this typing style is [deprecated](https://docs.python.org/3/library/typing.html#typing.List) (along with `Dict`, `Set` and friends) in favor of the standard `list` object.

So I would say it's up to you to choose: if we target 3.9+, we probably should move to the new style; otherwise, we should stick with the old one.

Reply:
That's good to know. Let's go with `list`. We will want to use `Expectation` instead of `Exception` 🙂

Reply:
👍 

> We will want to use Expectation instead of Exception 🙂

Indeed 🤦‍♂️